scripts/lib/libvirt/libvirt_setup.py

```diff
@@ -1,21 +1,29 @@
 from __future__ import print_function
-import xml.etree.ElementTree as ET
+
 import glob
-import libvirt
+import itertools as it
 import os
-from string import (Template, lowercase)
-import sys
+from string import lowercase
+from string import Template
 import subprocess
-import itertools as it
+import sys
+import xml.etree.ElementTree as ET
+
+import libvirt
+
 
 TEMPLATE_DIR = "{0}/templates".format(os.path.dirname(__file__))
 
 
 def libvirt_connect():
-    conn = libvirt.open("qemu:///system")
-    if not conn:
-        print("Failed to open connection to the hypervisor")
+
+    conn_uri = "qemu:///system"
+    try:
+        conn = libvirt.open(conn_uri)
+    except libvirt.libvirtError as e:
```

Comment:
This will happen due to many reasons like the libvirtd process not started, if the permissions are not appropriate or the URI definition is wrong etc. In future we should add this to logs.